### PR TITLE
feat: phpdoc_to_property_type - handle virtual types and null initialization, enable in php-highest CI job

### PR DESCRIPTION
**Pull request by keradus**
should unblock #8282

### CODE REVIEW

File: tests/Fixer/FunctionNotation/PhpdocToPropertyTypeFixerTest.php [hunk 1]
```diff
@@ -173,25 +173,45 @@ class Foo {
...
+
+        yield 'nullable type with default case 3' => [
+            '<?php class Foo { /** @var ?array */ private ?array $foo = []; }',
+            '<?php class Foo { /** @var ?array */ private $foo = []; }',
+        ];
+
+        yield 'nullable type with default case 4' => [
+            '<?php class Foo { /** @var ?Bar */ private ?Bar $foo = null; }',
+            '<?php class Foo { /** @var ?Bar */ private $foo; }',
+        ];
```
Comment (keradus) [inline tests/Fixer/FunctionNotation/PhpdocToPropertyTypeFixerTest.php:213]:
```suggestion
```

File: src/AbstractPhpdocToTypeDeclarationFixer.php [hunk 2]
```diff
@@ -95,6 +97,10 @@ protected function createConfigurationDefinition(): FixerConfigurationResolverIn
                 ->setAllowedTypes(['bool'])
                 ->setDefault(\PHP_VERSION_ID >= 8_00_00)
                 ->getOption(),
+            (new FixerOptionBuilder('map_types', 'Map of custom types, eg template types from PHPStan.'))
```
Comment (Wirone) [inline src/AbstractPhpdocToTypeDeclarationFixer.php:100]: IMHO it should be `types_map`, as `map_types` looks like an intent (or boolean flag, like the other options).

Comment (Wirone) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:100]: In general, I believe that PRs introducing changes in public contract of Fixer/rules shouldn't be merged without review.

Comment (keradus) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:100]: `types_map` vs `map_types` - open for both. no opinion

public contract - explicitly, [it is not that case](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/function_notation/phpdoc_to_param_type.rst#this-rule-is-experimental). Abstract parent class is [internal](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/AbstractPhpdocToTypeDeclarationFixer.php#L32) as well

Comment (Wirone) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:100]: #8293

**Pull request merged**